airbyte-webapp/src/components/ReleaseStageBadge/ReleaseStageBadge.tsx

```diff
@@ -27,6 +19,12 @@ interface ReleaseStageBadgeProps {
 }
 
 export const ReleaseStageBadge: React.FC<ReleaseStageBadgeProps> = ({ stage, small, tooltip = true }) => {
+  const { data } = useFreeConnectorProgramInfo();
+  const showEnrollmentUi = Boolean(data?.showEnrollmentUi) || true;
+  const { formatMessage } = useIntl();
+
+  const showFreeTag = showEnrollmentUi && (stage === "alpha" || stage === "beta");
```

Comment:
This requires a small update to `useFreeConnectorProgram`: we should only show these pills if the user's workspace is already enrolled.